File: python/structures/test/sorts/selection_sort_test.py
import structures.src.sorts.selection as selection_sort
from structures.src.util.sort_util import random_array, is_sorted
import structures.src.util.constants as const


def test_sorted():
    arr = random_array(const.ARRAY_SIZE)

    # make sure the starting array isn't sorted
    assert not is_sorted(arr)

    # sort the array
    result = selection_sort.sort(arr)

    # make sure it worked
    assert is_sorted(result)

File: ruby/test/sorts/selection_sort_test.rb
require_relative '../../src/sorts/selection_sort'
require_relative '../../src/util/util'
require_relative '../../src/util/constants'
require_relative '../util'

require 'test_helper'
require 'minitest/autorun'
require 'minitest/spec'

describe SelectionSort do
    describe 'when passed an array' do
        it 'should correctly sort it' do
            arr = TestUtil.random_array(Constants.ARRAY_SIZE)

            result = SelectionSort.sort(arr)

            Util.is_sorted(result).must_equal true
        end
    end
end
